Autopilot/AttitudeManager/AttitudeManager.c

```diff
@@ -604,26 +508,21 @@ int coordinatedTurn(float pitchRate, int rollAngle){
     return pitchRate;
 }
 
-int rollRateControl(float setpoint, float sensor){
-    rollPID = controlSignal(setpoint/SERVO_SCALE_FACTOR, sensor, ROLL);
-    return rollPID;
-}
-int pitchRateControl(float setpoint, float sensor){
-    pitchPID = controlSignal(setpoint/SERVO_SCALE_FACTOR, sensor, PITCH);
-    return pitchPID;
-}
-int yawRateControl(float setpoint, float sensor){
-    yawPID = controlSignal(setpoint/SERVO_SCALE_FACTOR, sensor, YAW);
-    return yawPID;
+// Type is both bit shift value and index of bit mask array
+uint8_t getControlValue(ctrl_type type) {
+    return (controlLevel & ctrl_mask[type]) >> type;
 }
 
-char getControlPermission(unsigned int controlMask, unsigned int expectedValue, char bitshift){
-    int maskResult = (controlMask & controlLevel);
-    return (maskResult >> bitshift) == expectedValue;
+void constrain(int16_t* input, int16_t min, int16_t max) {
```

Comment:
this is a pretty general function. should maybe go into a seperate file (utilities.c?)